### ISSUE TITLE
Totals of the cart are missing the currency

### ISSUE THREAD

**Issue by alexander-schranz** [Sylius/ShopApiPlugin#515]
Currently if you request a cart the totals look like this:

```json
"totals": {
    "total": 0,
    "items": 0,
    "taxes": 0,
    "shipping": 0,
    "promotion": 0
},
```

On all other places where I did see we response a price look like this.

```json
price: {
    "current": 2999,
    "currency": "GBP"
}
```

Using the price view.

I use mostly use a price react component in my application but this will not work for the totals. Is this by design so?

**Comment by mamazu** [comment]
You are right this is indeed a consistency break. But the idea behind it is the following:
All prices in the cart will have the same currency so specifying it would be redundant.

**Comment by alexander-schranz** [comment]
@mamazu thats right, but if it would be consistent it would be easier to create UI components based on this data.

**Comment by lchrusciel** [comment]
Totally makes sense. The question is, should each total be `price` view object? How it should be extended if we add `orginalPrice` or `discountedPrice` for instance?

Don't get me wrong, I'm totally in favor of this idea. I just don't see clear answers for my questions yet.

**Comment by alexander-schranz** [comment]
I would expect a instance of PriceView for everything:

```json
"totals": {
    "total": {
        "current": 2999,
        "currency": "GBP"
    },
    "items": {
        "current": 2999,
        "currency": "GBP"
    },
    "taxes": {
        "current": 0,
        "currency": "GBP"
    },
    "shipping": {
        "current": 0,
        "currency": "GBP"
    },
    "promotion": {
        "current": 0,
        "currency": "GBP"
    }
},
```

If originalPrice and discountedPrice are added they should also be a instance of PriceView.
